src/common/hooks/useCountyToZipMap.ts

```diff
@@ -1,51 +1,8 @@
-import { useEffect, useState } from 'react';
 import { singletonHook } from 'react-singleton-hook';
+import usePromise from './usePromise';
+import { importCountyToZipCodeMap } from 'common/data';
 
-const ZIPCODE_JSON_URL = '/data/county-zipcode.json';
-
-export interface CountyToZipMap {
-  [fips: string]: string[];
-}
-
-export interface UseCountyToZipMapReturn {
-  countyToZipMap: CountyToZipMap | undefined;
-  isLoading: boolean;
-}
-
-const init = {
-  countyToZipMap: undefined,
-  // initialize to true so client's don't have to handle case
-  // when not yet loading and not defined, since we kick off the load immediately
-  isLoading: true,
-};
-
-const useCountyToZipMapImpl = (): UseCountyToZipMapReturn => {
-  const [zipData, setZipData] = useState<CountyToZipMap | undefined>(
-    init.countyToZipMap,
-  );
-  const [isLoading, setIsLoading] = useState<boolean>(init.isLoading);
-
-  useEffect(() => {
-    const fetchZipData = async () => {
-      setIsLoading(true);
-      try {
-        const response = await fetch(ZIPCODE_JSON_URL);
-        const data = await response.json();
-        setZipData(data);
-      } catch (e) {
-        setZipData({});
-      } finally {
-        setIsLoading(false);
-      }
-    };
-    fetchZipData();
-  }, [setZipData, setIsLoading]);
-
-  return {
-    countyToZipMap: zipData,
-    isLoading,
-  };
-};
-
-const useCountyToZipMap = singletonHook(init, useCountyToZipMapImpl);
+const useCountyToZipMap = singletonHook({ pending: true }, () =>
+  usePromise(importCountyToZipCodeMap()),
+);
```

Comment:
FWIW- I tried a bit to create a helper to abstract away this boilerplate (using `singletonHook` and `usePromise` together), but had some trouble and I am in meetings most of the day and wanted to get this out.  I might try again later, though when monkeying with this stuff I kept tripping over https://reactjs.org/docs/hooks-rules.html